[TNT-181] 트레이니 일별 기록 화면 컴포넌트 구현

SeonJeongk opened the pull request:
## 📝 작업 내용

- #44 

- 디자인 시스템 `Chip` 컴포넌트를 구현했습니다. `ChipStyle`을 통해 칩의 색상을 관리하고 있습니다.`(BLUE, PINK)`
- 도메인의 `RecordType`모델에서 기록 유형을 정의하고, core:ui 모듈의 `RecordChip`모델을 통해 UI에서 표시될 텍스트와 아이콘 데이터를 관리하고 있습니다.
- 트레이니의 일별 기록 조회에 사용되는 `운동/식단/PT수업 카드` 컴포넌트를 구현했습니다
- PT수업 카드는 트레이너 홈 화면에도 사용됩니다. 왼쪽에 수업 확인 여부 체크 박스만 따로 붙여서 사용하시면 될 것 같습니다!
- 아래는 운동/식단 기록 카드 사용 예시 코드입니다.

``` kotlin
val chip = RecordChip.create(RecordType.DietType.BREAKFAST)
TnTRecordCard(
    style = ChipStyle.PINK,
    record = "닭가슴살 샐러드랑 고구마",
    tagText = chip.title,
    time = "오전 10:00",
    image = painterResource(co.kr.tnt.core.designsystem.R.drawable.img_default),
    modifier = Modifier.padding(5.dp),
    leadingEmoji = chip.emoji,
    feedbackCount = 2,
)
```

## 📸 실행 화면
#### Chip | 운동/식단 기록 카드 | PT 수업 카드  
| **Chip** | **운동/식단 기록 카드** | **PT 수업 카드** |
|----------|-----------------|--------------|
| <img src="https://github.com/user-attachments/assets/78734cab-ac46-46c4-a8c3-53126e4ccccb" width="150"/> | <img src="https://github.com/user-attachments/assets/68a630fe-c6d2-410f-ba88-a43e971cdc8f" width="200"/> | <img src="https://github.com/user-attachments/assets/abff977d-a40e-48d4-8d4e-9331c9703ad2" width="200"/> |



## 🙆🏻 리뷰 요청 사항
한 단어가 길어질 경우 줄바꿈이 발생하는 이슈가 있어 구글 이슈 트래커를 참고하며 수정 가능 여부 검토해보겠습니다
<img src="https://github.com/user-attachments/assets/ba5c564c-7de9-4565-9d7c-00a799b0c48c" width=200/>



## 👀 레퍼런스

SeonJeongk committed:
[TNT-181] move: Image 컴포넌트 파일 위치 수정

SeonJeongk committed:
[TNT-181] feat: Chip 컴포넌트 구현

SeonJeongk committed:
[TNT-181] refactor: Chip 컴포넌트 이름 수정 및 ChipStyle 추가

SeonJeongk committed:
[TNT-181] feat: 운동/식단 카드 컴포넌트 구현

SeonJeongk committed:
[TNT-181] feat: domain에 운동/식단/pt수업 type 추가

SeonJeongk committed:
[TNT-181] feat: PT 수업 카드 컴포넌트 구현

hoyahozz commented on core/designsystem/src/main/java/co/kr/tnt/designsystem/component/card/Card.kt:
```diff
@@ -0,0 +1,369 @@
+                    .clip(RoundedCornerShape(8.dp))
+                    .background(TnTTheme.colors.neutralColors.Neutral100)
+                    .padding(vertical = 12.dp)
+                    .clickable { onClick() },
```
```suggestion
                    .clickable(onClick = onClick),
```

SeonJeongk replied:
이놈의 람다.. 주의하겠습니다

hoyahozz commented on core/designsystem/src/main/java/co/kr/tnt/designsystem/component/card/Card.kt:
```diff
@@ -0,0 +1,369 @@
+    val recordBottomPadding = if (image == null) 8.dp else 0.dp
+    // 사진이 없고 피드백이 있으면 기록과 피드백 사이에 20.dp 간격 (20 - 8 = 12.dp)
+    val feedBackTopPadding = if (image == null) 12.dp else 0.dp
+    val imageBottomPadding = if (feedbackCount == null) 0.dp else 12.dp
```
움 요건 차라리 하단 컴포저블 선언 때 `if` 구문 넣어주는게 더 직관적일 것 같다는 생각이 드네요!

처음 봤을 때 왜 `20dp` 에서 `12dp` 를 빼지..? 하는 생각이 들었어요!

더불어 조건에 따라 `Spacer` 자체를 컴포저블 트리에 선언하지 않을거라 성능적으로도 더 나을 것 같구요..!

SeonJeongk replied:
그렇네요..! 주석이 길어질 때 눈치채고 바꿨어야 했는데 실수했습니다.. `if` 구문 넣어주도록 수정하겠습니다 👍

hoyahozz commented on core/designsystem/src/main/java/co/kr/tnt/designsystem/component/card/Card.kt:
```diff
@@ -0,0 +1,369 @@
+            image?.let {
+                Column {
+                    Image(
+                        painter = image,
```
`let` 블록 내의 `it` 을 사용해야 안정성이 100% 보장됩니다!

hoyahozz commented on core/ui/src/main/java/co/kr/tnt/ui/model/RecordChip.kt:
```diff
@@ -0,0 +1,62 @@
+import co.kr.tnt.domain.model.RecordType
+
+sealed class RecordChip(val title: String, val emoji: String?) {
+    class DietChip(
```
`data class` 사용해도 될 것 같습니다!

SeonJeongk replied:
`RecordChip` sealed interface로 수정하고 data class 사용하도록 수정했습니다!

hoyahozz commented on domain/src/main/java/co/kr/tnt/domain/model/RecordType.kt:
```diff
@@ -0,0 +1,22 @@
+package co.kr.tnt.domain.model
+
+sealed class RecordType {
+    sealed class DietType : RecordType() {
```
`MealType` 이라는 네이밍은 어떤가요?!

SeonJeongk replied:
움 저도 Meal이랑 많이 고민했는데 Meal이 더 알기 쉬운 이름이긴 하네요! 수정해두겠습니다

hoyahozz commented on domain/src/main/java/co/kr/tnt/domain/model/RecordType.kt:
```diff
@@ -0,0 +1,22 @@
+        data object CARDIO : ExerciseType()
+    }
+
+    sealed class PTSessionType : RecordType() {
```
```suggestion
    data class PTSessionType(val sessionCount: Int) : RecordType()
```

위 형태로 만들면, `create` 메소드에서 `sessionCount` 를 `nullable` 하게 설정하지 않아도 될 것 같아요.

항상 메소드나 클래스를 생성할 때 **'특정 상황에서만 사용되는 파라미터' 의 주입**은 피하는게 좋습니다!

SeonJeongk replied:
수정해습니다!! 주의하겠습니다💦

hoyahozz reviewed (approved):
고생하셨습니다~~~ 👍👍👍👍👍

저도 후딱 캘린더 만들어놓겠읍니다..

SeonJeongk committed:
[TNT-181] fix: chip의 style 네이밍 수정 및 리뷰 반영

- onClick 람다 제거
- let 블록 내 it 사용

SeonJeongk committed:
[TNT-181] fix: RecordChip class -> data class로 변경

- PTSessionType 수정 (create 메소드에서 sessionCount 를 nullable 하게 설정하지 않도록)

SeonJeongk committed:
[TNT-181] fix: Card padding 추가 방식 수정

SeonJeongk committed:
[TNT-181] feat: 운동 기록 카드 변경사항 반영

- 트레이니: showSessionRecordDetails로 "수업 기록 보기" 영역 보여지게 설정
- 트레이너: showSessionRecordCreation로 "PT 수업 기록 남기기" 영역 보여지게 설정

SeonJeongk committed:
[TNT-181] fix: create sessionCount 인자 삭제

SeonJeongk committed:
[TNT-181] fix: DietChip -> MealChip 이름 변경
